edgelet/iotedge/src/check/mod.rs

```diff
@@ -215,82 +200,89 @@ impl Check {
 
                 additional_info: AdditionalInfo::new(),
 
+                iothub_hostname,
                 settings: None,
                 docker_host_arg: None,
                 docker_server_version: None,
-                // iothub_hostname,
-                // device_ca_cert_path: None,
             })
         }))
     }
 
-    fn checks() -> [(&'static str, Vec<Box<dyn Checker>>); 1] {
-        /* Note: keep ordering consistant. Later tests may depend on earlier tests. */
-        [
-            (
-                "Configuration checks",
-                vec![
-                    Box::new(WellFormedConfig::default()),
-                    // Box::new(WellFormedConnectionString::default()),
-                    Box::new(ContainerEngineInstalled::default()),
-                    Box::new(Hostname::default()),
-                    Box::new(ParentHostname::default()),
-                    Box::new(ContainerResolveParentHostname::default()),
-                    Box::new(ConnectManagementUri::default()),
-                    Box::new(AziotEdgedVersion::default()),
-                    Box::new(HostLocalTime::default()),
-                    Box::new(ContainerLocalTime::default()),
-                    Box::new(ContainerEngineDns::default()),
-                    Box::new(ContainerEngineIPv6::default()),
-                    // Box::new(IdentityCertificateExpiry::default()),
-                    // Box::new(CertificatesQuickstart::default()),
-                    Box::new(ContainerEngineIsMoby::default()),
-                    Box::new(ContainerEngineLogrotate::default()),
-                    Box::new(EdgeAgentStorageMounted::default()),
-                    Box::new(EdgeHubStorageMounted::default()),
-                    Box::new(PullAgentFromUpstream::default()),
-                ],
-            ),
-            // ("Connectivity checks", {
-            //     let mut tests: Vec<Box<dyn Checker>> = Vec::new();
-            //     tests.push(Box::new(HostConnectDpsEndpoint::default()));
-            //     tests.extend(get_host_connect_upstream_tests());
-            //     tests.extend(get_host_container_upstream_tests());
-            //     tests
-            // }),
-        ]
-    }
+    pub fn print_list(aziot_bin: std::ffi::OsString) -> Result<(), Error> {
+        let mut all_checks: Vec<(String, Vec<CheckerMetaSerializable>)> = Vec::new();
 
-    pub fn possible_ids() -> impl Iterator<Item = &'static str> {
-        let result: Vec<&'static str> = Check::checks()
-            .iter()
-            .flat_map(|(_, section_checks)| section_checks)
-            .map(|check| check.id())
-            .collect();
+        // get all the aziot checks by shelling-out to aziot
+        {
+            let aziot_check_out = std::process::Command::new(aziot_bin)
+                .arg("check-list")
+                .arg("-j")
+                .output();
+
+            match aziot_check_out {
+                Err(_) => {
```

Comment:
Put the `Err` arm after the `Ok` arm please, to be consistent with the rest of the codebase.